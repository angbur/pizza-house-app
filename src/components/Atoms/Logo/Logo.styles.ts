import { getInitialTheme } from './../../Theme/initialTheme';
import styled from 'styled-components';

export const LogoBox = styled.div`
  height: 138px;
  background: ${getInitialTheme().palette.secondary};
  display: flex;
  justify-content: center;
  align-items: center;
  width: 114px;
  cursor: pointer;
`;
