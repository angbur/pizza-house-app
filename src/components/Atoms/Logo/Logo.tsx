import clsx from 'clsx';
import { useContext } from 'react';
import styled, { CSSProperties } from 'styled-components';
import LogoSVG from '../../../assets/image/logo.svg';
import { ColorVariants } from '../../Theme/theme.types';
import { ThemeContext } from '../../Theme/ThemeContext';

const logo = ['full', 'signet'] as const;
type LogoVariant = typeof logo[number];

export type LogoProps = {
  variant?: LogoVariant;
  background: ColorVariants;
  className?: string;
  style?: CSSProperties;
};

const LogoBox = styled.div`
  height: 138px;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 114px;
  cursor: pointer;
`;

const Logo = ({ variant = 'full', className, background, style }: LogoProps) => {
  const theme = useContext(ThemeContext);

  const backgroundStyles: CSSProperties = {
    background: theme.palette[background],
  };

  return (
    <LogoBox className={clsx(variant, className, 'logo')} style={{...backgroundStyles, ...style}}>
      <img src={LogoSVG} />
    </LogoBox>
  );
};

export default Logo;
