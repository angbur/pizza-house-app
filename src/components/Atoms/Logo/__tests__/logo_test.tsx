import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import Logo from '../Logo';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <Logo background='secondary' />
    </ThemeProvider>
  );
};

it('renders correctly logo', () => {
  const tree = renderer.create(<Wrapper />).toJSON();
  expect(tree).toMatchSnapshot();
});
