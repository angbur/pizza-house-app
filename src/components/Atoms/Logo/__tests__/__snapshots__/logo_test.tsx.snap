// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly logo 1`] = `
<div
  className="sc-bcXHqe hnPJle full logo"
  style={
    Object {
      "background": "#151515",
    }
  }
>
  <img
    src="logo.svg"
  />
</div>
`;
